frontend/src/metabase/hooks/use-confirm-leave-modal.ts

```diff
@@ -0,0 +1,78 @@
+import type { Location } from "history";
```

Comment:
Given that similar hooks exist already, out of curiosity what's the motivation for this?

Reply:
I've taken a couple of stabs at this but am struggling to make it work. The `onConfirmRouteLeave` hook you're referring to depends on a `confirm` callback that is changing some of the timing of when certain variables like `nextLocation` are available to the caller (I need this value to know what type of confirmation modal).

Politely, I'm convinced at this point my method is better/simpler than having to use both `useConfirmation` + `useConfirmOnRouteLeave` and avoids having to have hooks that return react components (`useConfirmation`). IMO we should refactor those components away, but I think that's more work than I'd like to take on before the holidays.